desktop/ui/src/main/java/org/datacleaner/widgets/tree/TableMouseListener.java

```diff
@@ -158,6 +162,24 @@ public void actionPerformed(ActionEvent e) {
                         _schemaTree.getDatastore(), columns));
                 popup.add(previewMenuItem);
 
+                final Datastore datastore = _schemaTree.getDatastore();
+                if (datastore instanceof UpdateableDatastore) {
+                    popup.addSeparator();
+
+                    final UpdateableDatastore updateableDatastore = (UpdateableDatastore) datastore;
+                    final WindowContext windowContext = injector.getInstance(WindowContext.class);
+                    final JMenuItem dropTableMenuItem = WidgetFactory.createMenuItem("Drop table", IconUtils.ACTION_DELETE);
+                    dropTableMenuItem.addActionListener(new ActionListener() {
+                        @Override
+                        public void actionPerformed(ActionEvent e) {
+                            final DropTableDialog dialog = new DropTableDialog(windowContext, updateableDatastore,
+                                    table, _schemaTree);
+                            dialog.open();
+                        }
+                    });
+                    popup.add(dropTableMenuItem);
+                }
+
```

Comment:
If possible, it would be nice to get some of this code out of this event handler. It is already waaaaayyyy too complex.